iceoryx_dust/include/iceoryx_dust/cxx/string_conversion.hpp

```diff
@@ -0,0 +1,44 @@
+// Copyright (c) 2022 by Apex.AI Inc. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+// SPDX-License-Identifier: Apache-2.0
+#ifndef IOX_DUST_STRING_CONVERSION_HPP
```

Comment:
What do you think about renaming this to `std_string_support.hpp`?

Currently the logger has a dependency to `std::string` in order to easily be able to log the STL type. This functionality could be move to this file and then something like `std_string_support.hpp` would fit better.

Reply:
@elBoberido That makes sense. I adapted the naming.